### PR TITLE
Add E2E test for spice search and chat functionality (OpenAI)

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Add E2E tests for `spice search` and `spice chat` functionality (OpenA) using [expect](https://core.tcl-lang.org/expect/index) (simple tool for automating interactive applications)
 - `s3://spiceai-public-datasets/tpcds/catalog_page/` is used to test `Utf8View` type  records with NULL values
 - `github:github.com/spiceai/spiceai/issues` covers `Utf8` type and real content characteristics (length variety, etc.)

Example test run: https://github.com/spiceai/spiceai/actions/runs/11950463815

## 🔨 Related Issues

Part of https://github.com/spiceai/spiceai/issues/3009

## 🤔 Concerns

1. OpenAI chat returns stable result for queries like `How many issues stored in database?` so test can include specific number instead of testing that chat request just completed.

### CODE REVIEW

File: test/models/spicepod.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    metadata:
+      instructions: Always provide reference links. Use the primary key as `issue_id` to generate reference links.
+      reference_url_template: https://github.com/spiceai/spiceai/issues/<issue_id>
+    params:
+      github_token: ${secrets:GITHUB_TOKEN}
+      github_query_mode: "search"
+    acceleration:
+      enabled: true
+      refresh_retry_max_attempts: 3
+      refresh_sql: "SELECT * FROM issues WHERE created_at='2024-11-09T22:47:45'"
```
Comment (sgrebnov) [inline test/models/spicepod.yml:24]: Embeddings can't currently handle NULL records, filter will be updated to include more records once it is fixed

File: test/models/spicepod.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          - id
+    
+  - from: s3://spiceai-public-datasets/tpcds/catalog_page/
+    name: catalog_page
+    description: Represents an e-commerce platform's catalog page metadata
+    params:
+      file_format: parquet
+    acceleration:
+      enabled: true
+      refresh_sql: SELECT * FROM catalog_page limit 1;
```
Comment (sgrebnov) [inline test/models/spicepod.yml:38]: Embeddings can't currently handle NULL records, filter will be updated to include more records once it is fixed

File: test/models/chat_01.exp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    expect {
+        " ERROR " {
+            # allow stream to complete
+            sleep 2
+            send_user "Error detected\n"
+            exit 1
+        }
+        "chat> " {
+            # end of model response
```
Comment (Sevenannn) [inline test/models/chat_01.exp:91]: Does these test only verify whether the chat is successfully initiated by the question, but doesn't verify the correctness of the result?

Comment (sgrebnov) [reply test/models/chat_01.exp:91]: Correct, in first iteration it is proposed to just verify that model responds to the query (the api works). If response is reliable we will then add check for specific response.

File: .github/workflows/e2e_test_ci_models.yml [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+                name: "Linux x64",
+                builder: "spiceai-runners",
+                runner: "ubuntu-latest",
+                target_os: "linux",
+                target_arch: "x86_64",
+                target_arch_go: "amd64"
+              }
+            ];
+
+            return context.eventName === 'pull_request' ? matrix.slice(0, 1) : matrix;
```
Comment (phillipleblanc) [inline .github/workflows/e2e_test_ci_models.yml]: If we aren't using the dynamic matrix, I'd prefer to just define the matrix inline on the build job

File: .github/workflows/e2e_test_ci_models.yml [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+  build:
+    # if: false
```
Comment (phillipleblanc) [inline .github/workflows/e2e_test_ci_models.yml]:
```suggestion
```

File: .github/workflows/e2e_test_ci_models.yml [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+          path: |
+            spice
+            spiced
+
+  test_openai_model:
+    name: 'openai model (${{ matrix.target.target_os }}-${{ matrix.target.target_arch }})'
+    runs-on: ${{ matrix.target.runner }}
+    needs:
+      - build
+      - setup-matrix
```
Comment (phillipleblanc) [inline .github/workflows/e2e_test_ci_models.yml]: Add timeout-minutes to each job

**Review by phillipleblanc — DISMISSED** [review]

File: .github/workflows/e2e_test_ci_models.yml (hunk 4 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Updated, thank you!

File: .github/workflows/e2e_test_ci_models.yml (hunk 6 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Added 30m for build and 10m for test execution, thank you!

File: .github/workflows/e2e_test_ci_models.yml (hunk 5 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Thank you

File: .github/workflows/e2e_test_ci_models.yml (hunk 4 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Example test run: https://github.com/spiceai/spiceai/actions/runs/11950463815

**Review by Sevenannn — APPROVED** [review]

**Pull request merged**